File: app/src/main/java/se/hactar/movieregister/repository/dto/Result.java
package se.hactar.movieregister.repository.dto;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

@SuppressWarnings("unused")
public class Result {
    @SerializedName("l")
    public String title;

    @SerializedName("id")
    public String id;

    @SerializedName("s")
    public String starring;

    @SerializedName("y")
    public String year;

    @SerializedName("q")
    public String type;

    @SerializedName("i")
    public final List<String> image = new ArrayList<>();

    @Override
    public String toString() {
        return "Result{" +
                "title='" + title + '\'' +
                ", id='" + id + '\'' +
                ", starring='" + starring + '\'' +
                ", year=" + year +
                ", type='" + type + '\'' +
                ", image=" + image +
                '}';
    }
}

File: app/src/main/java/se/hactar/movieregister/api/imdb/model/Result.kt
package se.hactar.movieregister.api.imdb.model


import com.google.gson.annotations.SerializedName

import java.util.ArrayList

class Result {
    @SerializedName("l")
    var title: String? = null

    @SerializedName("id")
    var id: String? = null

    @SerializedName("s")
    var starring: String? = null

    @SerializedName("y")
    var year: Int = 0

    @SerializedName("q")
    var type: String? = null

    @SerializedName("i")
    var images: List<String> = ArrayList()

    override fun toString(): String {
        return "Result(title=$title, id=$id, starring=$starring, year=$year, type=$type, images=$images)"
    }
}
